src/commons/mail-message-renderer.jsx

```diff
@@ -178,10 +261,103 @@ const _HtmlMessageRenderer = ({ msgId, body, parts, t }) => {
 		resizeObserver.observe(divRef.current);
 
 		return () => resizeObserver.disconnect();
-	}, [body, parts, msgId, contentToDisplay]);
+	}, [contentToDisplay, msgId, parts, showImage]);
 
 	return (
 		<div ref={divRef} className="force-white-bg">
+			{showBanner && !showExternalImage && (
+				<BannerContainer
+					orientation="horizontal"
+					mainAlignment="space-between"
+					crossAlignment="center"
+					padding={{ all: 'large' }}
+					height="58px"
+					background="#FFF7DE"
+					width="100%"
+				>
+					<Row
+						height="fit"
+						orientation="vertical"
+						display="flex"
+						wrap="nowrap"
+						mainAlignment="flex-start"
+						style={{
+							flexGrow: 1,
+							flexDirection: 'row'
+						}}
+					>
+						<Padding right="large">
+							<Icon icon="AlertTriangleOutline" color="warning" size="large" />
+						</Padding>
+						<Text overflow="break-word" size="small">
+							{t(
+								'message.external_images_blocked',
+								'External images have been blocked to protect you against potential spam'
+							)}
+						</Text>
+					</Row>
+					<Row
+						height="fit"
+						orientation="vertical"
+						display="flex"
+						wrap="nowrap"
+						mainAlignment="flex-end"
+						padding={{ left: 'small' }}
+						style={{
+							flexGrow: 1,
+							flexDirection: 'row'
+						}}
+					>
+						<StyledMultiBtn
+							background="transparent"
+							type="outlined"
+							label={t('label.view_images', 'VIEW IMAGES')}
+							color="warning"
+							onClick={() => {
+								setShowExternalImage(true);
+							}}
+							btnProps={{ size: 'fit', isSmall: true }}
+							dropdownProps={{
+								maxWidth: '500px',
+								width: 'fit'
+							}}
+							items={[
+								{
+									id: 'always-allow-address',
+									label: (
+										<Trans
+											i18nKey="label.always_allow_address"
+											defaults="Always allow from <strong>{{from}}</strong>"
```

Comment:
I'm little confuse here ,I guess in defaults instead of {{from}} there should be {{value}} because value contains the domain name.
@giuliano176  @zovomat  can you confirm this ,I'm little confuse right now

Reply:
looks alright, values is the object of variables to interpolate